src/nvim/statusline.c

```diff
@@ -867,6 +868,32 @@ void draw_tabline(void)
   redraw_tabline = false;
 }
 
+int build_statuscol_str(win_T *wp, bool setnum, bool wrap, linenr_T lnum, long relnum, int maxwidth,
+                        int fillchar, char *buf, stl_hlrec_t **hlrec, statuscol_T *stcp)
+{
+  if (setnum) {
+    set_vim_var_nr(VV_LNUM, lnum);
+    set_vim_var_nr(VV_RELNUM, relnum);
+  }
+  set_vim_var_bool(VV_WRAP, wrap);
+
+  StlClickRecord *clickrec;
+  char *stc = xstrdup(wp->w_p_stc);
+  int width = build_stl_str_hl(wp, buf, MAXPATHL, stc, "statuscolumn", OPT_LOCAL,
+                               fillchar, maxwidth, hlrec, &clickrec, stcp);
+  xfree(stc);
+
+  // Allocate and fill click def array if width has changed
+  if (wp->w_status_click_defs_size != (size_t)width) {
```

Comment:
It is so minor that I don't even need to open an issue, so I'll just write it here. Isn't `wp->w_statuscol_click_defs_size` correct?

Reply:
@luukvbaal does it make sense to include this change in one of your followup PRs?

Reply:
Not sure. The guard is only present in the alloc function. Having it here still avoids clearing and filling the click defs [unnecessarily](https://github.com/neovim/neovim/issues/21767).